packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -287,20 +309,32 @@ export async function sideloadAddIn(manifestPath: string, app?: OfficeApp, canPr
   if (isDesktop && app) {
     if (isSideloadingSupportedForDesktopHost(app)) {
       await registerAddIn(manifestPath);
-      sideloadFile = await generateSideloadFile(app, manifest, document);
+      if (app == OfficeApp.Outlook) {
+        sideloadFile = await getOutlookExePath();
+      } else {
+        sideloadFile = await generateSideloadFile(app, manifest, document);
+      }
     } else {
       throw new Error(`Sideload is not supported for ${app} on ${AppType.Desktop}.`);
     }
   } else {
-    if (platform && platform == AppType.Web && document === undefined) {
+    if (platform && platform == AppType.Web && document === undefined && app && isSideloadingSupportedForWebHost(app)) {
```

Comment:
You wouldn't need to check "app" here if done earlier. You can put the `isSideloadingSupportedForWebHost(app) check first and return the error if not, and then put the check for the document after it.

Reply:
You shouldn't need to check `platform && platform == AppType.Web) if you check for it earlier.